### ISSUE TITLE
Loss in training loops

### ISSUE THREAD

**Issue by fazelehh** [aidotse/LeakPro#239]
# Issue
## Problem Description

The last batch of each epoch is usually smaller than the others. One approach is to sum the losses and divide by the total number of samples.  

Currently, loss calculations in handlers sum the average loss of each batch and divide by the length of the dataloader.

**Comment by Kushagra7777** [comment]
Hello @fazelehh . I am a regular visitor of LeakPro repo and I was looking to make some contributions and recently came across this issue. I forked the main branch and made some first commits. I would like to carry it forward and put a closure on it but i am not sure how you guys are planning to use this feature. I would love to have a chat regarding this project. 
https://github.com/Kushagra7777/LeakPro/commit/b368726a46904bccf38e7e0725bda6175e67aa31
